src/main/java/org/influxdb/impl/InfluxDBImpl.java

```diff
@@ -97,6 +97,24 @@ public InfluxDBImpl(final String url, final String username, final String passwo
     this.adapter = moshi.adapter(QueryResult.class);
   }
 
+    InfluxDBImpl(final String url, final String username, final String password, final OkHttpClient.Builder client,
```

Comment:
I don't see any other way how to mock the InfluxDBService and the JsonAdapter (which will need to throw the Exceptions). The constructor has package visibility so that it can only be called from the same package (i.e. the Unit Test).